### PR TITLE
Include a sublibrary, Qubesdb_ipv4, which builds an ipv4 from a Qubes.DB

### PR DESCRIPTION
**Pull request by yomimono**
Qubesdb_ipv4 is intended for use with the `mirage` frontend tool as a
replacement for static or dynamic IPv4 configuration.  As QubesOS
doesn't provide a DHCP server for VMs and doesn't surface IP
configuration information before the VM exists, a nice interface for
statically configuring the interface with the information looked up from
QubesDB is required for MirageOS unikernels to be good network citizens
on QubesOS.

This module just wraps the `Static_ipv4` module from `tcpip` to build an `ipv4` with the supplied parameters from QubesDB.  It makes no attempt to monitor the database for changes or sync the unikernel's IP configuration with them.

### CODE REVIEW

File: lib/ipv4/qubesdb_ipv4.ml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Make(D: Qubes.S.DB) (Ethernet : V1_LWT.ETHIF) (Arp : V1_LWT.ARP) = struct
+  module IP = Static_ipv4.Make(Ethernet)(Arp)
+  include IP
+  let connect db ethif arp =
+    let (>>=?) ip f = match ip with
+      | None -> None
+      | Some s -> f s
+    in
+    let ip = D.read db "/qubes-ip" >>=? Ipaddr.V4.of_string in
+    let netmask = D.read db "/qubes-netmask" >>=? Ipaddr.V4.of_string in
```
Comment (talex5) [inline lib/ipv4/qubesdb_ipv4.ml]: According to https://github.com/talex5/qubes-mirage-firewall/issues/9#issuecomment-249525405, `/qubes-netmask` is a decoy and you should assume `255.255.255.255` in all cases.

**Comment by yomimono** [comment]
> I guess we should wait to merge this until Mirage 3 is out, right?

If you want `master` to be buildable with the current stuff in `opam-repository`, yes.  For testing integration, we can just as easily build against a branch -- just let me know what you'd prefer.

**Comment by talex5** [comment]
Yeah, let's keep it on a branch for now then.

File: lib/ipv4/qubesdb_ipv4.ml (hunk 1 above)
Comment (yomimono) [inline lib/ipv4/qubesdb_ipv4.ml]: The `routing` implementation in mirage-tcpip won't handle that very well currently.

**Comment by yomimono** [comment]
I can change the requested target branch of a PR, but apparently I can't request that a PR be merged to an entirely new branch.  It's up to you, `mirage-qubes-owner`!

**Comment by talex5** [comment]
OK, we have an ipv4 branch now.

**Comment by yomimono** [comment]
Thanks very much! I'll close this out, then.

File: lib/ipv4/qubesdb_ipv4.ml (hunk 1 above)
Comment (yomimono) [inline lib/ipv4/qubesdb_ipv4.ml]: It turns out that a /32 netmask not only works fine in current mirage-tcpip, but is required for running on GCE as well...

**Pull request closed without merging**